client/src/http/httpClient.js

```diff
@@ -0,0 +1,40 @@
+import { createClient } from './index.js';
+import { authService } from '../services/authService.js';
+import { accessTokenService } from '../services/accessTokenService.js';
+
+export const httpClient = createClient();
+
+httpClient.interceptors.request.use(onRequest);
+httpClient.interceptors.response.use(onResponseSuccess, onResponseError);
+
+function onRequest(request) {
+  const accessToken = localStorage.getItem('accessToken');
+
+  if (accessToken) {
+    request.headers['Authorization'] = `Bearer ${accessToken}`;
+  }
+
+  return request;
+}
+
+function onResponseSuccess(res) {
+  return res.data;
+}
+
+async function onResponseError(error) {
+  const originalRequest = error.config;
+
+  if (error.response.status !== 401) {
+    throw error;
+  }
+
+  try {
+    const { accessToken } = await authService.refresh();
```

Comment:
Ensure that the authService.refresh() method is correctly implemented and handles token refresh logic properly. If it fails, it should throw an error that can be caught in the catch block.